src/context/EASClientProvider.tsx

```diff
@@ -14,6 +16,16 @@ interface EASClientProviderProps {
 
 export const EASClientProvider = ({ children }: EASClientProviderProps) => {
   const [easClient, setEASClient] = useState<EASClientContextValue>(null);
+  const { push } = useRouter();
+  const { isConnected } = useAccount();
+
+  useEffect(() => {
+    if (!isConnected) {
+      push("/");
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isConnected]);
```

Comment:
does this logic have anything to do with `EAS`?
- if no, please move this code to a more appropriate provider
- if no appropriate provider exists, make a new one